addon/components/high-charts.js

```diff
@@ -39,40 +39,53 @@ export default Component.extend({
   didReceiveAttrs() {
     this._super(...arguments);
 
-    if (!(get(this, 'content') && get(this, 'chart'))) {
+    const { content, chart, mode } = this.getProperties('content', 'chart', 'mode');
+    if (!content || !chart) {
       return;
     }
 
-    let chart = get(this, 'chart');
     let noData = chart.get('noData');
```

Comment:
I know you didn't add this "noData" functionality but I think we should consider it when we refactor this didReceiveAttrs() code. What does the "noData" code do? When I go through the highcharts docs it doesn't seem to have a `remove()` method. The extra highcharts file no-data-to-display.js is needed to use this feature. @ahmadsoe do you know why this noData check is needed?

http://api.highcharts.com/highcharts#noData
http://www.highcharts.com/news/125-new-features-drilldown-and-no-data
http://jsfiddle.net/gh/get/jquery/1.7.2/highslide-software/highcharts.com/tree/master/samples/highcharts/no-data-to-display/no-data-pie/